### PR TITLE
feat: add webhook notifications

### PR DESCRIPTION
**Pull request by darookee**
### Added

- webhook notifications channel with multiple recipients

## Screenshots

image

### CODE REVIEW

**Comment by alexjustesen** [comment]
Thanks for the PR! I'm going to hold off reviewing/merging this until #736 is merged so we don't create extra work

File: app/Listeners/SpeedtestCompletedListener.php
Comment (DAcodedBEAT) [inline app/Listeners/SpeedtestCompletedListener.php:82]: Since consuming apps might require that the request body is structured in a specific way, should this body be configurable? (similar to something like https://docs.overseerr.dev/using-overseerr/notifications/webhooks#json-payload)

**Comment by masterwishx** [comment]
can it be used for Discord notify?

**Comment by alexjustesen** [comment]
> can it be merged now ?

Don't worry I haven't forgotten about this PR. Lots of moving pieces with trying to fix the time zone issue currently and then I should be able to come back to this.

**Comment by masterwishx** [comment]
@alexjustesen any news for this pr ?

**Comment by alexjustesen** [comment]
@darookee can you update your PR with the latest files and ensure tests pass?

**Comment by masterwishx** [comment]
@darookee can it be used for discord notification?

**Comment by darookee** [comment]
> @darookee can you update your PR with the latest files and ensure tests pass?

The tests passed locally but I'm not sure how to run the linting job locally, my IDE does not give any errors though :shrug:

**Comment by jonezy35** [comment]
I would LOVE to have this functionality. Is the plan to merge this?

**Comment by jonezy35** [comment]
@alexjustesen anything I can do to help make sure this is good to merge?

**Comment by alexjustesen** [comment]
I'm updating the PR as we speak to include recent changes. Need to do some 🤏 minor refactoring and then can merge it, eta prob later this weekend.

File: app/Filament/Pages/Settings/NotificationPage.php
Comment (alexjustesen) [inline app/Filament/Pages/Settings/NotificationPage.php:167]: Use `columnSpanFull()` because all components are full width.

Comment (alexjustesen) [reply app/Filament/Pages/Settings/NotificationPage.php:167]: I'll refactor later, fine for now.

**Review by alexjustesen — CHANGES_REQUESTED** [review]

**Comment by alexjustesen** [comment]
Merging into `add-webhook-notifications` to make edits easier.

**Comment by evan326** [comment]
like [masterwishx](https://github.com/masterwishx) asked twice, can/how this be used with discord?

**Comment by masterwishx** [comment]
> like [masterwishx](https://github.com/masterwishx) asked twice, can/how this be used with discord?

https://github.com/alexjustesen/speedtest-tracker/pull/754#issuecomment-1940377371

**Comment by darookee** [comment]
I don't know anything about discord webhooks but just looked it up in the discord docs and it looks like it does not accept requests without a specific json body. So no, with just _this_ MR it is not possible to call a discord webhook.

**Comment by darookee** [comment]
> That's why I posted link above where @alexjustesen said it's not working for now but will fix it in future, I hope it will be very soon...

The link points to @evan326's comment 😬 (at least for me), that's why I wanted to clarify.

**Comment by jonezy35** [comment]
> like [masterwishx](https://github.com/masterwishx) asked twice, can/how this be used with discord?

I'd be happy to see if I can contribute to add the discord functionality! Life has been super hectic but if I get some time I'll see if I can do some work and open up a PR.

For now I've been using this. Works well and uses the API that speedtest exposes.

**Comment by jonezy35** [comment]
> like [masterwishx](https://github.com/masterwishx) asked twice, can/how this be used with discord?

Discord functionality is added in PR #1196. As long as everything is good and it's merged Discord functionality will be up and running.

**Pull request merged**
